noir-projects/aztec-nr/aztec/src/encrypted_logs/log_assembly_strategies/default_aes128/note.nr

```diff
@@ -0,0 +1,557 @@
+}
+
+// Ok, so we have the following bytes:
+// eph_pk_sign, header_ciphertext, ciphertext:
+// Let lbwop = 1 + 48 + |ct| // aka log bytes without padding
+fn get_arr_of_size__log_bytes_without_padding<let CT: u32>(_ct: [u8; CT]) -> [u8; 1 + 48 + CT] {
+    [0; 1 + 48 + CT]
+}
+
+// Recall:
+//   lbwop := 1 + 48 + |ct| // aka log bytes without padding
+// We now want to pad b to the next multiple of 31, so as to "fill" fields.
+// Let p be that padding.
+// p = 31 * ceil(lbwop / 31) - lbwop
+//   = 31 * ((lbwop + 30) // 31) - lbwop
+//     (because ceil(x / y) = (x + y - 1) // y ).
+fn get_arr_of_size__log_bytes_padding<let LBWOP: u32>(
+    _lbwop: [u8; LBWOP],
+) -> [u8; (31 * ((LBWOP + 30) / 31)) - LBWOP] {
+    [0; (31 * ((LBWOP + 30) / 31)) - LBWOP]
+}
+
+// |log_bytes| = 1 + 48 + |ct| + p // aka log bytes (with padding)
+// Recall:
+//   lbwop := 1 + 48 + |ct|
+//   p is the padding
+fn get_arr_of_size__log_bytes<let LBWOP: u32, let P: u32>(
+    _lbwop: [u8; LBWOP],
+    _p: [u8; P],
+) -> [u8; LBWOP + P] {
+    [0; LBWOP + P]
+}
+
+// The return type is pasted from the LSP's expectation, because it was too difficult
+// to match its weird way of doing algebra. It doesn't know all rules of arithmetic.
+// PT is the plaintext length.
+pub(crate) fn get_arr_of_size__log_bytes_padding__from_PT<let PT: u32>() -> [u8; ((((((PT + (16 - (PT % 16))) + 49) + 30) / 31) * 31) - ((PT + (16 - (PT % 16))) + 49))] {
+    let full_pt = get_arr_of_size__full_plaintext::<PT>();
+    let pt_aes_padding = get_arr_of_size__plaintext_aes_padding(full_pt);
+    let ct = get_arr_of_size__ciphertext(full_pt, pt_aes_padding);
+    let lbwop = get_arr_of_size__log_bytes_without_padding(ct);
+    let p = get_arr_of_size__log_bytes_padding(lbwop);
+    p
+}
+
+// The return type is pasted from the LSP's expectation, because it was too difficult
+// to match its weird way of doing algebra. It doesn't know all rules of arithmetic.
+pub(crate) fn get_arr_of_size__log_bytes__from_PT<let PT: u32>() -> [u8; (((PT + (16 - (PT % 16))) + 49) + ((((((PT + (16 - (PT % 16))) + 49) + 30) / 31) * 31) - ((PT + (16 - (PT % 16))) + 49)))] {
+    let full_pt = get_arr_of_size__full_plaintext::<PT>();
+    let pt_aes_padding = get_arr_of_size__plaintext_aes_padding(full_pt);
+    let ct = get_arr_of_size__ciphertext(full_pt, pt_aes_padding);
+    let lbwop = get_arr_of_size__log_bytes_without_padding(ct);
+    let p = get_arr_of_size__log_bytes_padding(lbwop);
+    let log_bytes = get_arr_of_size__log_bytes(lbwop, p);
+    log_bytes
+}
+
+/********************************************************/
+// End of disgusting arithmetic on generics
+/********************************************************/
+
+// TODO: it feels like this existence check is in the wrong place. In fact, why is it needed at all? Under what circumstances have we found a non-existent note being emitted accidentally?
+fn assert_note_exists<Note, let N: u32>(context: PrivateContext, note: Note)
```

Comment:
Yeah this is tech debt, tracking in https://github.com/AztecProtocol/aztec-packages/issues/8589